tests/docFetcher.test.ts

```diff
@@ -0,0 +1,148 @@
+import { it, expect, beforeEach, vi } from 'vitest';
+import { ContentType } from '../src/constants.ts';
+import { clearDocCache, fetchDocPage } from '../src/docFetcher.ts';
+
+const mocks = vi.hoisted(() => ({
+  rmAll: vi.fn(),
+  get: {
+    info: vi.fn(),
+  },
+  put: vi.fn(),
+  fetch: vi.fn(),
+  clearCache: vi.fn(),
+}));
+
+vi.mock('../src/fetchService.ts', async (importOriginal) => ({
+  ...(await importOriginal<typeof import('../src/fetchService.ts')>()),
+  fetchService: {
+    fetch: mocks.fetch,
+    clearCache: mocks.clearCache,
+  },
+}));
+
+vi.mock('cacache', async (importOriginal) => ({
+  ...(await importOriginal<typeof import('cacache')>()),
+  get: {
+    info: mocks.get.info,
+  },
+  put: mocks.put,
+  rm: {
+    all: mocks.rmAll,
+  },
+}));
+
+vi.mock('node:crypto', async (importOriginal) => ({
```

Comment:
How come we're mocking this? Hashing is a pure function, it's a better test if we use inputs that produce a known hash and check that the result matches.

Reply:
No idea, if I'm being honest at this stage I haven't really been reading what the tests were doing but only focused on porting them to the new tools and structures.

I mainly did this to avoid an even bigger diff, but I imagine we'll go through all the tests as we update the implementation.